src/openlifu/io/LIFUUart.py

```diff
@@ -293,6 +294,34 @@ def list_vcp_with_vid_pid(self):
                 return port.device
         return None
 
+    def reopen_after_reset(self, retries=5, delay=1.0):
+        """
+        Attempt to reopen the serial port after a device reset or disconnection.
+        Returns True if reconnected, False otherwise.
+        """
+        try:
+            if self.serial and self.serial.is_open:
+                with suppress(Exception):
+                    self.serial.close()
+            self.serial = None
+            self.port = None
+            for _ in range(retries):
+                time.sleep(delay)
+                port = self.list_vcp_with_vid_pid()
+                if port:
+                    self.port = port
+                    try:
+                        self.connect()
+                        log.info("Reconnected to UART on %s after reset.", self.port)
+                        return True
+                    except Exception as e:
+                        log.warning("Reconnect attempt failed: %s", e)
+            log.error("Failed to reconnect UART after reset.")
+            return False
+        except Exception as e:
+            log.error("reopen_after_reset() error: %s", e)
+            return False
```

Comment:
Better not to [swallow up arbitrary exceptions like this](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/broad-exception-caught.html), because it can unintentionally hide future bugs and make debugging difficult.

Best is to catch the specific exception you are trying to catch

An also ok approach would be to re-raise the exception here instead of `return False`. (but then what's the point of these lines? the python traceback will already show that the exception occured in `reopen_after_reset()` so there is no need to log that fact manually)

(Same goes for the other line above that swallows arbitrary exceptions)